src/frontend/components/sidebar.tsx

```diff
@@ -1,14 +1,107 @@
-import React from 'react'
-import MenuOptions from './menu-options'
+"use client";
+import React, { useEffect, useState } from "react";
+import MenuOptions from "./menu-options";
+import { useWindowSize } from "../hooks/useWindowSize";
+import { FilePlus, Menu } from "lucide-react";
+import {
+  Sheet,
+  SheetClose,
+  SheetContent,
+  SheetDescription,
+  SheetFooter,
+  SheetHeader,
+  SheetTitle,
+  SheetTrigger,
+} from "@/components/ui/sheet";
 
-type Props = {}
+type Props = {
+  isOpen: boolean;
+  toggleSidebar: () => void;
+};
```

Comment:
I see you added these props but it doesnt seem they are used anywhere. Also when we call `Sidebar` on `app/page.tsx`, it doesn't seem like we pass these props.